scripts/move-product.sh

```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
+
+# === CLI ARGS ===
+
+DRY_RUN=false
+PRODUCT_NAME=""
+
+usage() {
+  echo "Usage: $0 [--dry-run] [--product=<productName>]"
+  exit 1
+}
+
+for arg in "$@"; do
+  case $arg in
+  --dry-run)
+    DRY_RUN=true
+    ;;
+  --product=*)
+    PRODUCT_NAME="${arg#*=}"
+    ;;
+  --product)
+    shift
+    PRODUCT_NAME="$1"
+    ;;
+  *)
+    usage
+    ;;
+  esac
+done
+
+if [ -z "$PRODUCT_NAME" ]; then
+  echo "📦 Let's move your product!"
+  read -rp "Enter the name of the product: " PRODUCT_NAME
+  echo ""
+fi
+
+if [ -z "$PRODUCT_NAME" ]; then
+  echo "❌ Product name is required."
+  exit 1
+fi
+
+# === Color codes ===
+RESET='\033[0m'
+GRAY='\033[2m'
+RED='\033[31m'
+GREEN='\033[32m'
+YELLOW='\033[33m'
+BLUE='\033[34m'
+CYAN='\033[36m'
+
+# === Color function ===
+color() {
+  local text="$1"
+  local color="$2"
+  case "$color" in
+  gray) echo -e "${GRAY}${text}${RESET}" ;;
+  red) echo -e "${RED}${text}${RESET}" ;;
+  green) echo -e "${GREEN}${text}${RESET}" ;;
+  yellow) echo -e "${YELLOW}${text}${RESET}" ;;
+  blue) echo -e "${BLUE}${text}${RESET}" ;;
+  cyan) echo -e "${CYAN}${text}${RESET}" ;;
+  *) echo "$text" ;;
+  esac
+}
+
+# === Paths & Helpers ===
+PRODUCT_DIR="src/products/${PRODUCT_NAME}"
+LAYOUT_DIR="src/templates/layouts/${PRODUCT_NAME}"
+QUERY_FILE="src/data/queries/${PRODUCT_NAME}.ts"
+FORMATTED_TYPE_FILE="src/types/formatted/${PRODUCT_NAME}.ts"
+CODEOWNERS_FILE=".github/CODEOWNERS"
+
+run_or_echo() {
+  if $DRY_RUN; then
+    echo -e "${GRAY}   [dry-run] $*${RESET}"
+  else
+    eval "$@"
+  fi
+}
+
+log() {
+  echo -e "${BLUE}🔧 $1${RESET}"
+}
+
+warn() {
+  echo -e "${YELLOW}⚠️  $1${RESET}"
+}
+
+success() {
+  echo -e "${GREEN}✅ $1${RESET}"
+}
+
+error() {
+  echo -e "${RED}❌ $1${RESET}"
+}
+
+update_all_imports() {
+  log "🔎 Scanning project for import references to '$(color "$PRODUCT_NAME" cyan)'"
+
+  # Parallel arrays: old paths and their replacements
+  searches=(
+    "@/templates/layouts/${PRODUCT_NAME}"
+    "@/data/queries/${PRODUCT_NAME}"
+    "@/data/queries/tests/${PRODUCT_NAME}"
+    "@/types/formatted/${PRODUCT_NAME}"
+    "@/mocks/${PRODUCT_NAME}.mock"
+  )
+
+  replacements=(
+    "@/products/${PRODUCT_NAME}"
+    "@/products/${PRODUCT_NAME}/query"
+    "@/products/${PRODUCT_NAME}/query.test"
+    "@/products/${PRODUCT_NAME}/formatted-type"
+    "@/products/${PRODUCT_NAME}/mock"
+  )
+
+  # Loop through both arrays by index
+  i=0
+  while [ $i -lt ${#searches[@]} ]; do
+    search="${searches[$i]}"
+    replace="${replacements[$i]}"
+    i=$((i + 1))
+
+    log "Replacing imports from $(color "$search" yellow) → $(color "$replace" green)"
+
+    # Escape slashes in search and replace patterns
+    search_escaped=$(printf '%s\n' "$search" | sed 's/[\/&]/\\&/g')
+    replace_escaped=$(printf '%s\n' "$replace" | sed 's/[\/&]/\\&/g')
+
+    # Find and replace with word boundaries
+    grep -rl --include="*.ts*" --include="*.js*" "$search" ./src | while read -r file; do
```

Comment:
Technically, things could get a little silly if we've got a JSON file with a `@/templates/layouts/event` or whatever. 😅 🤷